Core/Dialog/SavedataParam.cpp

```diff
@@ -790,35 +793,33 @@ u32 SavedataParam::LoadNotCryptedSave(SceUtilitySavedataParam *param, u8 *data,
 	return 0;
 }
 
-void SavedataParam::LoadSFO(SceUtilitySavedataParam *param, const std::string& dirPath) {
+bool SavedataParam::LoadSFO(SceUtilitySavedataParam *param, const std::string& dirPath) {
 	ParamSFOData sfoFile;
-	std::string sfopath = dirPath+"/" + SFO_FILENAME;
-	PSPFileInfo sfoInfo = pspFileSystem.GetFileInfo(sfopath);
-	if (sfoInfo.exists) {
-		// Read sfo
-		std::vector<u8> sfoData;
-		if (pspFileSystem.ReadEntireFile(sfopath, sfoData) >= 0) {
-			sfoFile.ReadSFO(sfoData);
+	std::string sfopath = dirPath + "/" + SFO_FILENAME;
+	std::vector<u8> sfoData;
+	if (pspFileSystem.ReadEntireFile(sfopath, sfoData) >= 0) {
+		sfoFile.ReadSFO(sfoData);
 
-			// copy back info in request
-			strncpy(param->sfoParam.title,sfoFile.GetValueString("TITLE").c_str(),128);
-			strncpy(param->sfoParam.savedataTitle,sfoFile.GetValueString("SAVEDATA_TITLE").c_str(),128);
-			strncpy(param->sfoParam.detail,sfoFile.GetValueString("SAVEDATA_DETAIL").c_str(),1024);
-			param->sfoParam.parentalLevel = sfoFile.GetValueInt("PARENTAL_LEVEL");
-		}
+		// copy back info in request
+		strncpy(param->sfoParam.title,sfoFile.GetValueString("TITLE").c_str(),128);
+		strncpy(param->sfoParam.savedataTitle,sfoFile.GetValueString("SAVEDATA_TITLE").c_str(),128);
+		strncpy(param->sfoParam.detail,sfoFile.GetValueString("SAVEDATA_DETAIL").c_str(),1024);
+		param->sfoParam.parentalLevel = sfoFile.GetValueInt("PARENTAL_LEVEL");
 	}
+	return true;
```

Comment:
Hm, shouldn't this return false when we couldn't read the entire file?

-[Unknown]